lib/jellygrinder/ll_client.ex

```diff
@@ -74,27 +73,33 @@ defmodule Jellygrinder.LLClient do
     {:noreply, state}
   end
 
-  defp request_track_manifest(state) do
-    state.base_uri
-    |> Map.update!(:path, &Path.join(&1, state.track_manifest_name))
-    |> request("media playlist", state.name, state.parent)
-  end
+  defp create_track_manifest_query(%{latest_partial: nil} = _state), do: ""
 
-  defp request_partial(partial, state, retries \\ @max_single_partial_request_retries)
+  defp create_track_manifest_query(%{latest_partial: latest_partial} = _state) do
+    [last_msn, last_part] =
+      latest_partial.name
+      |> then(
+        &Regex.run(~r/^muxed_segment_(\d+)_\w*_(\d+)_part.m4s$/, &1, capture: :all_but_first)
+      )
+      |> Enum.map(&String.to_integer/1)
+
+    {msn, part} =
+      if last_part == @partials_per_segment - 1,
+        do: {last_msn + 1, 0},
+        else: {last_msn, last_part + 1}
```

Comment:
Which one is easier to understand?
```suggestion
    last_part = last_part + 1
    msn = last_msn + div(last_part,@partials_per_segment - 1)
    part = rem(last_part,@partials_per_segment - 1)
```

Reply:
We've decided together with @Karolk99 that we're getting rid of it anyway